Modules/Module_Processing/Module_Population/xASL_adm_DeleteManyTempFiles.m

```diff
@@ -13,34 +13,76 @@ function xASL_adm_DeleteManyTempFiles(x)
 % -----------------------------------------------------------------------------------------------------------------------------------------------------
 % EXAMPLE:      xASL_adm_DeleteManyTempFiles(x);
 % __________________________________
-% Copyright 2015-2023 ExploreASL
+% Copyright 2015-2024 ExploreASL
 
-Files2Del = {'ATT_BiasField.nii' 'Mask_Template.nii' 'Mean_CBF_Template.nii' 'PseudoCBF.nii' 'RawTemplate.nii' 'VascularArtifact_Template.nii' 'mean_PWI_Clipped.nii' 'SliceGradient_extrapolated.nii' 'FoV.nii'};
+Files2DelNativeStructural = {}; % (currently no files here)
+Files2DelNativeASL = {'ATT_BiasField.nii' 'Mask_Template.nii' 'Mean_CBF_Template.nii' 'PseudoCBF.nii' 'RawTemplate.nii' 'VascularArtifact_Template.nii' 'mean_PWI_Clipped.nii' 'SliceGradient_extrapolated.nii' 'FoV.nii'};
+Files2DelStandardStructural = {'rT1_ORI_' 'rT1_' 'rFLAIR_' '(m|)rc\dT1_'};
+Files2DelStandardASL = {'noSmooth_M0_' 'mean_control_' 'PWI_' 'SliceGradient_' 'SNR'};
```

Comment:
If we delete PWI after completing the entire ASL module. Then if we try to re-run ASL module from the "CreateMask" part, it crashes as it needs PWI. Do we really need to delete PWI?

Reply:
We don't need to delete anything, but the more we delete the more data we save. We don't delete the native space PWI, only the standard space PWI.

The only reason I could think of, is that we may create the standard space PWI from ASL4D warped to standard space. If we re-create PWI by transforming the native space PWI, we may get more interpolation artifacts.

Reply:
This is the whole point why we now only delete such "temporary files" for completed modules. You give an example that you want for an incompleted module. So this situation will not occur. Unless you want to rerun part of a module, this is currently only possible if you haven't ran the population module.

I would keep it this way. We have the following cases in which we run the full population module
1. a module is not completed yet, so we don't delete the file (this solves your issue)
2. a module is fully completed, so we delete the file (you cannot redo half of a module, you always have to redo the whole module. unless you avoid running the population module.) Note that this is the case for many things. Even without deleting temporary files, you cannot arbitrary remove a few lock files and rerun a module; we have dependencies within a module. In fact we solve this for the case that you have only the first 2 lock files of a module done, if you then rerun a module it will also delete these first 2 lock files and repeat the full module.

3. If you are an advanced user and you want to repeat a module partly (e.g. for comparing different settings), you just don't run the population yet. The population module is the "end/finalizing" module.

Reply:
I mostly talked about the scenario of an advanced user re-running half of a module after removing lock-files. But I agree that people shouldn't do this. So all good :D I'm resolving this and merging!